### PR TITLE
Use Commit Proto

### PR DESCRIPTION
**Pull request by CaseyHillers**
1. Exported protos from cocoon_service in a separate protos file. Exporting from the main cocoon_service file resulted in import `dart:mirror` which Flutter will not run.
2. Made cocoon_service pubspec compatible with the Flutter project (just decremented collections down a minor version).
3. Updated the Flutter side to use the Commit proto instead of temporary variables

### CODE REVIEW

File: app_dart/lib/protos.dart [hunk 1]
```diff
@@ -0,0 +1,5 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+export 'src/model/proto/protos.dart';
```
Comment (dnfield) [inline app_dart/lib/protos.dart]: nit: newline at EOF

File: app_flutter/lib/commit_box.dart [hunk 2]
```diff
@@ -11,22 +12,10 @@ import 'package:flutter/material.dart';
...
 /// will close it.
 class CommitBox extends StatefulWidget {
-  // TODO(chillers): convert to use commit model
-  const CommitBox(
-      {Key key,
-      @required this.message,
-      @required this.avatarUrl,
-      @required this.author})
-      : super(key: key);
+  const CommitBox({Key key, @required this.commit}) : super(key: key);
```
Comment (dnfield) [inline app_flutter/lib/commit_box.dart]: `assert(commit != null), super(key: key);`

File: app_flutter/lib/commit_box.dart [hunk 3]
```diff
@@ -70,15 +60,15 @@ class CommitOverlayContents extends StatelessWidget {
   CommitOverlayContents({
     Key key,
     @required this.parentContext,
-    @required this.widget,
+    @required this.commit,
```
Comment (dnfield) [inline app_flutter/lib/commit_box.dart]: Assert that these are all not null (or at least that commit is not null).

File: app_flutter/lib/commit_box.dart [hunk 4]
```diff
@@ -70,15 +60,15 @@ class CommitOverlayContents extends StatelessWidget {
...
     @required this.closeCallback,
   }) : super(key: key);
 
   /// The parent context that has the size of the whole screen
   final BuildContext parentContext;
 
-  /// The parent widget that contains state variables
-  final CommitBox widget;
+  /// The commit of the parent widget to show information from
```
Comment (dnfield) [inline app_flutter/lib/commit_box.dart]: This doc string is a bit hard to parse.

Maybe `The commit data to display in the overlay.`?

File: app_flutter/lib/commit_box.dart [hunk 5]
```diff
@@ -115,11 +105,12 @@ class CommitOverlayContents extends StatelessWidget {
                 ListTile(
                   leading: CircleAvatar(
                     radius: 25.0,
-                    backgroundImage: NetworkImage(widget.avatarUrl),
+                    backgroundImage: NetworkImage(commit.authorAvatarUrl),
                     backgroundColor: Colors.transparent,
                   ),
-                  title: Text(widget.message),
-                  subtitle: Text(widget.author),
+                  // TODO(chillers): Show commit message here instead
```
Comment (dnfield) [inline app_flutter/lib/commit_box.dart]: TODOs should have bugs filed and a link to the bug.  `// TODO(chillers): Show commit message here instead: https://github.com/flutter/flutter/issues/999999`

Comment (CaseyHillers) [reply app_flutter/lib/commit_box.dart]: Thank you Dan! I learned a new thing :)

File: app_flutter/lib/commit_box.dart (hunk 4 above)
Comment (CaseyHillers) [reply app_flutter/lib/commit_box.dart]: Thank you. That is much clearer and the meaning I wanted to convey
